File: langs/python/prime.py
import sys
import math

def sieve(size):
    if (n <= 1):
        return [False] * size
    sieve_array = [True] * size
    sieve_array[0] = sieve_array[1] = False
    root = int(math.sqrt(size)) + 1
    for i in range(2, root):
        if not sieve_array[i]:
            continue
        for j in range(i * i, size, i):
            sieve_array[j] = False
    return sieve_array

def count_prime(n):
    return sieve(n + 1).count(True)

n = int(sys.argv[1]) if (len(sys.argv) >= 2) else 10000000
print(count_prime(n))

File: langs/go/prime.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
)

func sieve(size int) (sieveArray []bool) {
	sieveArray = make([]bool, size)
	for i := 2; i < size; i++ {
		sieveArray[i] = true
	}
	root := int(math.Sqrt(float64(size))) + 1
	for i := 2; i < root; i++ {
		if sieveArray[i] {
			for j := i * i; j < size; j += i {
				sieveArray[j] = false
			}
		}
	}
	return
}

func countPrime(n int) (count int) {
	count = 0
	for _, v := range sieve(n + 1) {
		if v {
			count++
		}
	}
	return
}

func main() {
	n := 10000000
	if len(os.Args) >= 2 {
		i, err := strconv.Atoi(os.Args[1])
		if err == nil {
			n = i
		}
	}
	fmt.Println(countPrime(n))
}
